# Hypothesis Testing Problem Statement

### 1. Anda adalah seorang Manager yang baru bekerja di Supermarket X. Berdasarkan data history di tahun 2020, rata-rata usia pembeli di Supermarket Anda berusia 20 tahun ke atas. Anda ingin melakukan study di bulan-bulan ini setelah Anda bergabung, berapa banyak proporsi pembeli yang berusia 20 tahun ke atas. Untuk itu, Anda mengambil random sample 500 pengunjung yang melakukan pembelian di Supermarket Anda. Tuliskan Null Hypothesis dan Alternative Hypothesis dari studi yang Anda lakukan.


#### *Null Hypothesis* / Hipotesis Nol (Ho)

Ho: Tidak ada perbedaan proporsi pembeli yang berusia 20 tahun ke atas di Supermarket X antara tahun 2020 dan bulan-bulan ini.

#### *Alternative Hypothesis* / Hipotesis Alternatif (Ha)

Ha: Ada perbedaan proporsi pembeli yang berusia 20 tahun ke atas di Supermarket X antara tahun 2020 dan bulan-bulan ini.

### 2. Terhadap 500 orang tersebut, Anda ingin melakukan uji 1 sample-test untuk memverifikasi hipotesis yang telah Anda tulis di jawaban No. 1. Berikut adalah distribusi usia dari 500 random sample tersebut. (Buka file age_distribution.txt)
### a. Uji Hypothesistesting apakah yang akan Anda lakukan? Ambil asumsi normality dalam hal ini.
### b. Berapakah besaran test statistic dari uji hipotesis pada poin a)
### c. Berapakah besaran p- value-nya?
### d. Bagaimana kesimpulan dari hypothesis testing poin a) dan b) Apakah Anda akan me- reject atau accept Null Hypothesis-nya?

#### a) Uji hipotesis yang akan digunakan adalah Uji T-Satu Sampel. Uji T-Satu Sampel digunakan untuk menguji apakah rata-rata populasi sama dengan suatu nilai tertentu. Pada studi ini, nilai tertentunya yang diuji sebagai hipotesis adalah 20 tahun.

#### b) Besaran *test statistic* dari uji hipotesis pada poin a) adalah sebagai berikut:

In [ ]:
# Mengimport fungsi ttest_1samp dari modul scipy.stats untuk melakukan uji t-test satu sampel
from scipy.stats import ttest_1samp

# Menggunakan library numpy untuk membantu mengolah data yang digunakan dalam studi ini
import numpy as np

# Data yang digunakan dalam studi ini diambil dari file 'age_distribution.txt' (dinyatakan sebagai 'data')
data = np.loadtxt('age_distribution.txt')

# Menghitung besaran t-statistic dan p-value
t_statistic, p_value = ttest_1samp(data, 20, alternative='two-sided')

print(f'Besaran test statistic dari Uji T-Satu Sampel pada studi ini adalah', t_statistic)

Besaran test statistic dari Uji T-Satu Sampel pada studi ini adalah 2.6343913433586814


#### c) Besaran p- value adalah sebagai berikut:

In [ ]:
print(f'Besaran p-value pada studi ini adalah', p_value.round(4))

Besaran p-value pada studi ini adalah 0.0087


#### d. Bagaimana kesimpulan dari hypothesis testing poin a) dan b) Apakah Anda akan me- reject atau accept Null Hypothesis-nya?

In [ ]:
# Menginisialisasi alpha, tingkat signifikansi uji hipotesis (0.05 = 5%)
alpha = 0.05

# Melakukan uji hipotesis. Jika nilai p-value lebih besar dari alpha, maka hipotesis nol diterima. Sebaliknya, jika nilai p-value lebih kecil dari alpha, maka hipotesis nol ditolak.
if p_value > alpha:
  print (f'Null Hypothesis / Hipotesis Nol (Ho) diterima. Tidak ada perbedaan proporsi pembeli yang berusia 20 tahun ke atas di Supermarket X antara tahun 2020 dan bulan-bulan ini.')
else:
  print (f'Null Hypothesis / Hipotesis Nol (Ho) ditolak. Ada perbedaan proporsi pembeli yang berusia 20 tahun ke atas di Supermarket X antara tahun 2020 dan bulan-bulan ini.')

Null Hypothesis / Hipotesis Nol (Ho) ditolak. Ada perbedaan proporsi pembeli yang berusia 20 tahun ke atas di Supermarket X antara tahun 2020 dan bulan-bulan ini.


Pada studi ini, nilai p-value adalah 0.0087. Nilai ini lebih kecil dari alpha = 0.05. Oleh karena itu, hipotesis nol ditolak.